File: samples/client/wordnik-api/php/tests/WordsApiTest.php
<?php

require_once 'BaseApiTest.php';

date_default_timezone_set('America/Los_Angeles');

class WordsApiTest extends BaseApiTest {

  public function testSearchWords() {
    $res = $this->wordsApi->searchWords('tree');
    $this->assertEquals('tree', $res->searchResults[0]->word);
    $this->assertNotEquals(0, $res->totalResults);
  }

  public function testGetWordOfTheDay() {
    $res = $this->wordsApi->getWordOfTheDay();
    $this->assertNotEquals(null, $res);
  }

  public function testReverseDictionary() {
    $res = $this->wordsApi->reverseDictionary('hairy');
    $this->assertNotEquals(0, $res->totalResults);
    $this->assertNotEquals(0, count($res->results));
  }

  public function testGetRandomWords() {
    $res = $this->wordsApi->getRandomWords();
    $this->assertEquals(10, count($res));
  }

  public function testGetRandomWord() {
    $res = $this->wordsApi->getRandomWord();
    $this->assertNotEquals(null, $res);
  }

}
?>

File: samples/client/wordnik-api/python3/tests/WordsApiTest.py
#!/usr/bin/env python

import sys
import unittest
import urllib.request, urllib.error, urllib.parse
import json

from BaseApiTest import BaseApiTest

sys.path = ['./'] + sys.path
from wordnik import *


class WordsApiTest(BaseApiTest):

    def testSearchWords(self):
        res = self.wordsApi.searchWords('tree')
        assert res, 'null search result'
        assert res.searchResults[0].word == 'tree', 'word should be "tree"'
        assert res.totalResults != 0, 'should not have 0 results'

    def testGetWordOfTheDay(self):
        res = self.wordsApi.getWordOfTheDay()
        assert res, 'null wordOfTheDay result'

    def testReverseDictionary(self):
        res = self.wordsApi.reverseDictionary("hairy")
        assert res, 'null reverseDictionary result'
        assert res.totalResults != 0, 'should not have 0 results'
        assert len(res.results) != 0, 'should not have 0 results'

    def testGetRandomWords(self):
        res = self.wordsApi.getRandomWords()
        assert res, 'null getRandomWords result'
        assert len(res) == 10, 'should get 10 random words'

    def testGetRandomWords(self):
        res = self.wordsApi.getRandomWords()
        assert res, 'null getRandomWord result'

    def testGetRandomWord(self):
        res = self.wordsApi.getRandomWords()
        assert res, 'null getRandomWord result'


if __name__ == "__main__":
    unittest.main()
